dbt/scripts/run_iasworld_data_tests.py

```diff
@@ -904,25 +982,12 @@ def create(
         )
 
         # Extract dbt vars
-        run_vars = get_key_from_run_results("args", run_results_filepath)[
-            "vars"
-        ]
-        var_year_start = run_vars.get("data_test_iasworld_year_start")
-        var_year_end = run_vars.get("data_test_iasworld_year_end")
-
-        # If dbt vars weren't set on the command line, the defaults won't exist
-        # in run_results.json, so we have to parse them from the dbt project
-        # config
-        if not var_year_start or not var_year_end:
-            with open("dbt_project.yml") as project_fobj:
-                project = yaml.safe_load(project_fobj)
-            var_year_start = (
-                var_year_start
-                or project["vars"]["data_test_iasworld_year_start"]
-            )
-            var_year_end = (
-                var_year_end or project["vars"]["data_test_iasworld_year_end"]
-            )
```

Comment:
I factored this logic out into a dedicated function so that we can reuse it for extracting values for SNS topic variables.